src/Features/Core/Portable/GenerateMember/GenerateDefaultConstructors/AbstractGenerateDefaultConstructorsService.AbstractCodeAction.cs

```diff
@@ -71,6 +69,28 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
                     statements: default,
                     baseConstructorArguments: baseConstructorArguments);
             }
+
+            private static Accessibility DetermineAccessibility(IMethodSymbol baseConstructor, INamedTypeSymbol classType)
+            {
+                if (baseConstructor.ContainingType.IsAbstractClass() && !classType.IsAbstractClass())
+                    return Accessibility.Public;
+
+                if (classType.IsSealed)
+                {
+                    // remove protected as it makes no sense in a sealed type.
+                    switch (baseConstructor.DeclaredAccessibility)
+                    {
+                        case Accessibility.Protected:
+                            return Accessibility.Public;
+                        case Accessibility.ProtectedAndInternal:
+                        case Accessibility.ProtectedOrInternal:
+                            return Accessibility.Internal;
+                    }
```

Comment:
for my own curiosity- I didn't see private protected in http://sourceroslyn.io/#Microsoft.CodeAnalysis/Symbols/Accessibility.cs
Is it not valid here?  Or does it need to be changed to internal as well?

Reply:
private-protected is called ProtectedAndInternal internally :)